core/imds/src/test/java/software/amazon/awssdk/imds/Ec2MetadataTest.java

```diff
@@ -121,9 +125,8 @@ public void get_AmiId_onMetadataResource_401Error_throws() throws IOException {
         stubFor(get(urlPathEqualTo(AMI_ID_RESOURCE)).willReturn(aResponse().withBody("{}").withStatus(401)));
 
         Ec2Metadata ec2Metadata = Ec2Metadata.builder().endpoint(URI.create("http://localhost:8080")).build();
-        String data = ec2Metadata.get("/latest/meta-data/ami-id");
-
-        assertThat(data).isNull();
+        MetadataResponse metadataResponse = ec2Metadata.get("/latest/meta-data/ami-id");
+        assertThat(metadataResponse).isNull();
```

Comment:
Returning null seems weird here. Any reason we wouldn't throw an exception? I'd think we'd never want get() to return null.